### ISSUE TITLE
failing to install a git+ssh dependency

### ISSUE THREAD

**Issue by andreineculau** [pnpm/pnpm#989]
### Code to reproduce the issue:

Unfortunately I cannot reproduce with a `git+ssh://git@github.com:user/foo.git#v0.0.1`, but only with an internal git server, so I'm gonna lay down descriptively all the public details I can share.

`bar.git` has its `package.json` as simple as

```json
{
  "private": true,
  "name": "bar",
  "version": "0.0.1",
  "dependencies": {
    "foo": "git+ssh://git@server.internal:some/path/foo.git#v0.0.1"
  }
```

while `foo.git` at its `v0.0.1` tag has its `package.json` as simple as

```json
{
  "private": true,
  "name": "foo",
  "version": "0.0.1",
  "dependencies": {
    "ajv": "4.11.8",
    "lodash": "4.17.4"
  },
  "scripts": {}
}
```

The "bisect" test was

```sh
$ cd bar
$ for f in 1.22.0 1.23.0 1.23.1 1.23.2 1.24.0 0 1.24.0 1 1.24.0 2 1.24.0 3 1.24.0 1.24.1 1.24.2 1.24.3 1.25.0 1.25.1; do echo "123 DELIMITER 123"; git clean -xdf .; rm -rf ~/.pnpm_store; echo "npx pnpm@$f i"; npx pnpm@$f i; touch pnpm-debug.log; mv pnpm-debug.log ../pnpm-debug.$f.log; done 2>&1 >../pnpm-repro-git-ssh.txt
```

All the generated logs from the "bisect" above are available on keybase https://keybase.pub/andreineculau/pnpm-repro-git-ssh/ *as zip files* (ignore the log files)

### Expected behavior:

Installation to succeed on all versions.

### Actual behavior:

See 20180110113131.zip

* `1.22.0 - 1.23.1 OK`
* `1.23.2 - 1.25.1 FAIL: pnpm i` - with "ERROR  EISDIR: illegal operation on a directory, read"
* `1.25.0 FAIL: pnpm i` - with no error

NOTE: the test is not stable though, both in regards to the silent error above, but also in regards to having `pnpm i` actually succeed (exit code 0) (and getting `FAIL: missing node_modules` output from my test). I could get that on 1.25.1 now and then, but consistently when calling pnpm directly (or via `npx pnpm` (no version)) --- See 20180110120815.zip

### Additional information:

 - `node -v` prints: 9.3.0
 - Windows, OS X, or Linux?: OSX

**Comment by andreineculau** [comment]
FWIW tried without `npx`, and I couldn't get the `FAIL: missing node_modules`, so even that is not "consistent", although in my shell, running `pnpm i` with `1.25.1` I consistently get no error, exit code 0 and no node_modules folder
See 20180110123855.zip

**Comment by zkochan** [comment]
Seems like the error happens inside [dint](https://github.com/zkochan/dint) (when calling `dint.from()`).

**Comment by zkochan** [comment]
@andreineculau I don't really see where's the issue. Maybe you could try to generate an integrity file for the repo from the logs. It easy, you just pass the location of the directory to `dint.from()`: https://github.com/zkochan/dint/blob/master/example.js

**Comment by zkochan** [comment]
This is most probably fixed in [v1.30.1](https://github.com/pnpm/pnpm/releases/tag/v1.30.1)

**Comment by andreineculau** [comment]
close but not cigar 🎉 

it fails now with `Error: ENOENT: no such file or directory, open '/Users/andrei/.pnpm-store/2/git@server.internal+some/path/foo.git/<hash>_stage/lib/atex-api-json/session-v1.js'`

if I `rm -rf node/modules/foo` and run `pnpm i` again, I get the ENOENT message basically for all the files in the `src` and `lib` folder (there are other folders in that archive, but pnpm doesn't complain about them - dunno if it's timing issue i.e. it exists before reaching them).

when I look in the pnpm store, I can see `/Users/andrei/.pnpm-store/2/git@server.internal+some/path/foo.git/<hash>/package/node_modules/lib/atex-api-json/session-v1.js` and every other file just fine.

while it seems obvious to me that ENOENT errors are swallowed by dint, I have looked a bit into it, and it looks like all of its `readfile` calls are all successful. but if not dint, where is the error coming from?

**Comment by zkochan** [comment]
ok, I know where the error is :man_facepalming: 

We rename the folder before the integrity is generated.

https://github.com/pnpm/package-requester/blob/master/src/packageRequester.ts#L402-L432

The fix is easy. I don't think it will make performance worse, either

**Comment by andreineculau** [comment]
close*r* but still no cigar :)

so now i consistently get *all* the ENOENT errors for src and lib folder

**Comment by zkochan** [comment]
oh, indeed. That option won't help because `integrity.json` is created anyway, only without the checksums.

anyway, fix will be ready today

**Comment by zkochan** [comment]
@andreineculau this should be fixed in `v1.30.2`

**Comment by andreineculau** [comment]
ohh, i should have checked my email or this thread. I came to confirm the fix (with a monkey-patched package-requester installation in pnpm's node_modules at 1.30.1).

i confirm also with a clean install of 1.30.2 :tada:
